### PR TITLE
Preserve check constraint during column move

### LINKED ISSUE #17920: Moving column with empty default value will replace it with ''
### Describe the bug

When you try to move a column where the default value is an empty string, the generated sql will replace the empty string with ''
Example
```sql
`ALTER TABLE `my_table` CHANGE `uniqid` `uniqid` VARCHAR(23) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '\'\'' AFTER `id` `
```

### To Reproduce

Create a table with at least 2 columns, one with varchar, and default value empty
change the order of the column
the default value is now ''

### Expected behavior

No change in default value

### Screenshots

If applicable, add screenshots to help explain the bug.

### Server configuration

- Operating system: linux
- Web server:
- Database version: 10.6.11-MariaDB-1:10.6.11+maria~ubu2004
- PHP version:8.0.25 
- phpMyAdmin version:5.2.0

### Client configuration

- Browser: firefox
- Operating system: windows 10

### Additional context

Add any other context about the bug here.

### LINKED ISSUE #18006: Moving columns causes the default uuid() value to get quotes around it
### Describe the bug

When I set a default value for a column that is `UUID` to `uuid()`, and use PHPMyAdmin to move it in the list of columns, and then use `show create table X`, I see that the `uuid()` would become `'uuid()'` and if I move it one more time, it becomes `'''uuid()'''`.

### To Reproduce

- Create a table with at least 3 columns
- Set two columns data types to be UUID
- Set default values for those two columns to `uuid()`
- Then use `Move column` feature to change the order of columns
  - Select the table
  - Go to `Structure` tab
  - Use `Move column` link (if you can't find it, search it inside the page)
- Then use `show create table X`
- You should see quotes around the default value
- If you move columns multiple times, then those quotes will increase in number

### Expected behavior

The default value `uuid()` should remain as is when we move columns (change the order of columns)

### Screenshots

From left hand (default state) we get to the right hand after moving columns around

Screenshot from 2023-01-07 16-19-15

### Server configuration

- Operating system:
- Web server:
- Database version:
- PHP version: MariaDB
- phpMyAdmin version: latest

### Client configuration

- Browser: Chrome
- Operating system:

### Additional context

Add any other context about the bug here.

### LINKED ISSUE #18962: Move columns with ENUM() & DEFAULT causes invalid SQL
### Describe the bug

Move columns fails to generate a valid `ALTER TABLE` statement when used on an `ENUM` field with a default value.

### To Reproduce

```sql
CREATE TABLE `test` (
  `id` int(11) NOT NULL,
  `enum` enum('yes','no') NOT NULL DEFAULT 'no',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci
```
1. Go to "Structure"
2. Click on "Move columns"
3. Drag `enum` above `id`
4. Click Go
5. Error: 
>  \# 1067 - Invalid default value for 'enum'

```sql
ALTER TABLE `test` 
CHANGE `enum` `enum` ENUM('yes','no') 
CHARACTER SET utf8 COLLATE utf8_unicode_ci NOT NULL 
DEFAULT '\'no\'' FIRST
```
Note the extraneous double quoting of the `DEFAULT 'no'`

### Expected behavior

The move columns feature should produce valid SQL for ENUM fields with a default value.

### Screenshots

image

image

### Server configuration

- Operating system: Rocky 8
- Web server: Apache
- Database version:  10.4.32-MariaDB-
- PHP version: 8.1.26
- phpMyAdmin version: 5.2.1-1

### PR DESCRIPTION
**Pull request by MoonE**
### Description
Fixes #17920, fixes #18006, fixes #18962
Closes #19009

See #13592, #14371, #16224
- [x] Any new functionality is covered by tests.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
This looks promissing

**Comment by MoonE** [comment]
[Here](https://github.com/MoonE/phpmyadmin/commit/ec436ac55161aed99009517c9348cfd9795d2e30) is the version when merging QA_5_2 -> master

File: js/src/table/structure.js [hunk 1]
```diff
@@ -322,9 +322,15 @@ AJAX.registerOnload('table/structure.js', function () {
         designerModalPreviewModal.addEventListener('shown.bs.modal', () => {
             const modalBody = designerModalPreviewModal.querySelector('.modal-body');
             const $form = $('#move_column_form');
+            const serialized = $form.serialize();
+            if (serialized === $form.data('serialized-unmoved')) {
+                modalBody.innerHTML = '';
+                return;
+            }
```
Comment (MoonE) [inline js/src/table/structure.js:329]: Don't make a request when nothing was changed. The controller now would return an error in this case. Before, an invalid alter table statement was returned without any column changes.

**Review by williamdes — APPROVED** [review]
Thank you !

**Pull request merged**